modules/ROOT/pages/raise-error-component-reference.adoc

```diff
@@ -4,23 +4,90 @@ include::_attributes.adoc[]
-`raise-error` components and custom types.
+Once you declare a custom namespace by using it in a `raise-error`, you can use it in other `raise-error` components and custom types.
 
-The following example flow produces an `ACCOUNT:INSUFFICIENT_FUNDS` error when a
-transfer amount surpasses the available balance, preventing the transfer from
-taking place. Additionally, it generates a dynamic description by providing an
-expression with failure details.
 
-.Example: XML Configuration of the Flow
+== Examples
+
+The following example of a driving lessons platform checks that the user is over 16 years old to drive. The flow raises a `PRECONDITIONS:INCORRECT_AGE` error when the user enters an age lower than 16 years.
+
+* The Choice router components adds conditional processing to the flow.
+* If the number value is lower than 16 in the request `http://localhost:8081/test?age=15`, the Raise Error component raises the error type `PRECONDITIONS:INCORRECT_AGE` and description `Minimum age of 16 required to drive`.
+* Then, a Logger component logs the error message `Minimum age required to drive is 16 years old` in the On Error Continue component flow whose type is `PRECONDITIONS:INCORRECT_AGE`.
+* If the number value is greater than 16 in the request `http://localhost:8081/test?age=17`, a Logger component returns the default message `User age above 16 years. Allowed to drive`.
+* If no age value parameter is passed in the request `http://localhost:8081/test`, a Logger component returns the message `The parameter age is missing`.
+
+XML Configuration of the Flow:
+[source,xml,linenums]
+----
+<http:listener-config name="HTTP_Listener_config">
+		<http:listener-connection host="0.0.0.0" port="8081" />
+	</http:listener-config>
+	<flow name="raise-error-example-flow">
+		<http:listener config-ref="HTTP_Listener_config" path="/test"/>
+		<choice>
+			<when expression="#[message.attributes.queryParams.age &lt; 16]">
+				<raise-error type="PRECONDITIONS:INCORRECT_AGE" description="Minimum age of 16 required to drive" />
+			</when>
+			<otherwise >
+				<logger level="INFO" message="User age above 16 years. Allowed to drive"/>
+			</otherwise>
+		</choice>
+		<error-handler >
+			<on-error-continue enableNotifications="true" logException="true" type="PRECONDITIONS:INCORRECT_AGE">
+				<logger level="INFO" message="Minimum age to drive is 16 years old"/>
+			</on-error-continue>
+			<on-error-continue enableNotifications="true" logException="true"  type="MULE:EXPRESSION">
+				<logger level="INFO" message="The parameter age is missing"/>
+			</on-error-continue>
+		</error-handler>
+	</flow>
+----
+
+The following example assumes that a flow produces an `ACCOUNT:INSUFFICIENT_FUNDS` error when a transfer amount surpasses the available balance, preventing the transfer from taking place. Additionally, it generates a dynamic description by providing an expression with failure details.
```

Comment:
Maybe you can add an introduction here: 

Consider an API that returns an account balance when making a request to `https://unsecurebank.com/balance`.
The following example, makes a request to this API and stores the value in the `balance` variable, then it compares it with `payload.amount` to determine if the operation is possible.
The Choice router produces an `ACCOUNT:INSUFFICIENT_FUNDS` error when a transfer amount surpasses the available balance, preventing the transfer from taking place. Additionally, it generates a dynamic description by providing an expression with failure details.